src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.hpp

```diff
@@ -1537,125 +1566,99 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_WriteThreadNetw
 #else
-            uint64_t activeTimestamp  = activeDataset.mActiveTimestamp;
+        uint64_t activeTimestamp  = activeDataset.mActiveTimestamp;
 #endif
-            err = encoder.Encode(activeTimestamp);
-        }
+        err = encoder.Encode(activeTimestamp);
     }
     break;
 
     case ThreadNetworkDiagnostics::Attributes::PendingTimestamp::Id: {
-        err = CHIP_ERROR_INCORRECT_STATE;
-        if (otDatasetIsCommissioned(mOTInst))
-        {
-            otOperationalDataset activeDataset;
-            otError otErr = otDatasetGetActive(mOTInst, &activeDataset);
-            VerifyOrExit(otErr == OT_ERROR_NONE, err = MapOpenThreadError(otErr));
+        otOperationalDataset activeDataset;
+        otError otErr = otDatasetGetActive(mOTInst, &activeDataset);
+        VerifyOrReturnError(otErr == OT_ERROR_NONE, MapOpenThreadError(otErr));
 #if OPENTHREAD_API_VERSION >= 219
-            uint64_t pendingTimestamp = (activeDataset.mPendingTimestamp.mSeconds << 16) |
-                (activeDataset.mPendingTimestamp.mTicks << 1) | activeDataset.mPendingTimestamp.mAuthoritative;
+        uint64_t pendingTimestamp = (activeDataset.mPendingTimestamp.mSeconds << 16) |
+            (activeDataset.mPendingTimestamp.mTicks << 1) | activeDataset.mPendingTimestamp.mAuthoritative;
 #else
-            uint64_t pendingTimestamp = activeDataset.mPendingTimestamp;
+        uint64_t pendingTimestamp = activeDataset.mPendingTimestamp;
 #endif
-            err = encoder.Encode(pendingTimestamp);
-        }
+        err = encoder.Encode(pendingTimestamp);
     }
     break;
 
     case ThreadNetworkDiagnostics::Attributes::Delay::Id: {
-        err = CHIP_ERROR_INCORRECT_STATE;
-        if (otDatasetIsCommissioned(mOTInst))
-        {
-            otOperationalDataset activeDataset;
-            otError otErr = otDatasetGetActive(mOTInst, &activeDataset);
-            VerifyOrExit(otErr == OT_ERROR_NONE, err = MapOpenThreadError(otErr));
-            uint32_t delay = activeDataset.mDelay;
-            err            = encoder.Encode(delay);
-        }
+        otOperationalDataset activeDataset;
+        otError otErr = otDatasetGetActive(mOTInst, &activeDataset);
+        VerifyOrReturnError(otErr == OT_ERROR_NONE, MapOpenThreadError(otErr));
+        uint32_t delay = activeDataset.mDelay;
+        err            = encoder.Encode(delay);
     }
     break;
 
     case ThreadNetworkDiagnostics::Attributes::SecurityPolicy::Id: {
-        err = CHIP_ERROR_INCORRECT_STATE;
-
-        if (otDatasetIsCommissioned(mOTInst))
-        {
-            otOperationalDataset activeDataset;
-            otError otErr = otDatasetGetActive(mOTInst, &activeDataset);
-            VerifyOrExit(otErr == OT_ERROR_NONE, err = MapOpenThreadError(otErr));
+        otOperationalDataset activeDataset;
+        otError otErr = otDatasetGetActive(mOTInst, &activeDataset);
+        VerifyOrReturnError(otErr == OT_ERROR_NONE, MapOpenThreadError(otErr));
```

Comment:
Is there a way to factor this code out of several switch cases? Can be done as a follow-up.

Reply:
Ok I will think of something and I'll push something soon